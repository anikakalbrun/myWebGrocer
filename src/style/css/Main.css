.main {
    background-image: url('../images/OFC16_Cafe_018_1600.png');
    height: 100%;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
}