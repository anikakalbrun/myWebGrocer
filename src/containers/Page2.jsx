import React, {Component} from 'react';
import { Field, reduxForm } from 'redux-form';
import {func, bool} from 'prop-types';

import Button from '../../src/components/Button.jsx';
import FormField from '../components/FormField';
import DateField from '../components/DateField';
import Wrapper from '../components/Wrapper';
import post from '../utils/post'
import {required, alf, maxLength30, maxLength60, minLength2, alf_space_dash, email} from '../utils/formValidation';
import {setPageNumber} from '../actions'
import '../style/css/Page.css'

export class Page2 extends Component {

    static propTypes = {
        submitting: bool.isRequired,
        invalid: bool.isRequired,
        handleSubmit: func.isRequired
    };

    render() {
        const {handleSubmit, pristine, submitting} = this.props;
        return (
            <Wrapper innerContainerClass='inverse default large'>
                <div className='row'>
                    <form className="form-horizontal" onSubmit={handleSubmit}>
                        <div className='col-xs-12 col-md-6'>
                            <Field name="firstName"
                                   component={FormField}
                                   label="First Name"
                                   placeholder="John"
                                   autoComplete='given-name'
                                   validate={[required, alf, minLength2, maxLength60]}/>

                            <Field name="dob"
                                   component={DateField}
                                   label="Date of Birth"
                                   placeholder="30/01/1975"
                                   showYearDropdown = {true}
                                   validate={[required]}/>

                            <Field name="email"
                                   component={FormField}
                                   label="Email Address"
                                   placeholder="john.doe@gmail.com"
                                   type='email'
                                   autoComplete='email'
                                   validate={[required, email]}/>
                        </div>

                        <div className='col-xs-12 col-md-6'>
                            <Field name="surname"
                                   component={FormField}
                                   label="Surname"
                                   placeholder="Doe"
                                   autoComplete='family-name'
                                   validate={[required, alf_space_dash, minLength2, maxLength60]}/>

                            <Field name="nationality"
                                   component={FormField}
                                   label="Nationality"
                                   placeholder="Irish"
                                   autoComplete='country-name'
                                   validate={[required, alf, minLength2, maxLength30]}/>

                            <Field name="occupation"
                                   component={FormField}
                                   label="Occupation"
                                   placeholder="Front End Developer"
                                   validate={[required, alf_space_dash, minLength2, maxLength60]}/>
                        </div>

                        <Button type='submit'
                                disabled={pristine || submitting}
                                text='Submit'
                                btnClassName='button--green pull-right m-r-15'/>
                    </form>
                </div>
            </Wrapper>
        );
    }
}

Page2 = reduxForm({
    form: 'RegistrationForm',
    destroyOnUnmount: true,
    onSubmit: post,
    url: '/users',
    callback: setPageNumber,
    pageNumber: 3
})(Page2);

export default Page2;
