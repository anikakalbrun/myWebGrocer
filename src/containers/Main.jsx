import React, {Component} from 'react';
import {connect} from "react-redux";
import {number, func} from 'prop-types';

import Page1 from '../components/Page1';
import Page2 from '../containers/Page2';
import Page3 from '../components/Page3';
import {setPageNumber} from '../actions';
import '../style/css/Main.css';

const getHeight = (innerHeight) => innerHeight - 91;

export class Main extends Component {

    static propTypes = {
        pageNumber: number.isRequired,
        setPageNumber: func.isRequired
    };

    state = {
        pageNumber: getHeight(window.innerHeight)
    };

    componentDidMount() {
        window.addEventListener("resize", this.setComponentHeight);
        this.setComponentHeight();
    };

    componentWillUnmount() {
        window.removeEventListener("resize", this.setComponentHeight);
    }

    setComponentHeight = () => {
        this.setState({minHeight: getHeight(window.innerHeight)})
    };

    goToSecondPage = () => {
        this.props.setPageNumber(2)
    };

    render() {
        const {pageNumber} = this.props;
        return (
            <main className='main'>
                <div style={{minHeight: this.state.minHeight, height: 'auto', overflow: 'auto'}}>
                    {pageNumber === 1 && <Page1 goToSecondPage={this.goToSecondPage}/>}
                    {pageNumber === 2 && <Page2/>}
                    {pageNumber === 3 && <Page3/>}
                </div>
            </main>
        );
    }
}

function mapStateToProps(state) {
    return {
        pageNumber: state.pageReducer.pageNumber
    }
}

export default connect(mapStateToProps, {setPageNumber})(Main)