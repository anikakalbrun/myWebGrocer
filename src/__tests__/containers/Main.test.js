import React from 'react';

import {Main} from '../../containers/Main';

let wrapper;
const setPageNumber = jest.fn();
const props = {
    setPageNumber: setPageNumber,
};

describe('Main', () => {

    it('setComponentHeight works fine and is being called by windows resize', () => {
        window.innerHeight = 800;
        wrapper = shallow(<Main pageNumber={1} {...props}/>);
        window.innerHeight = 1200;
        window.dispatchEvent(new Event('resize'));
        wrapper.update();
        expect(wrapper.find('.main > div').props().style).toMatchSnapshot();
    });

    it('goToSecondPage calls the action creator setPageNumber', () => {
        wrapper = shallow(<Main pageNumber={1} {...props}/>);
        wrapper.instance().goToSecondPage();
        expect(setPageNumber).toHaveBeenCalledWith(2);
    });

    it('renders the Page1 component when pageNumber=1', () => {
        wrapper = shallow(<Main pageNumber={1} {...props}/>);
        expect(wrapper.find('Page1').length).toEqual(1);
    });

    it('renders the Page2 component when pageNumber=2', () => {
        wrapper = shallow(<Main pageNumber={2} {...props}/>);
        expect(wrapper.props().children).toMatchSnapshot();
    });

    beforeEach(() => {
        wrapper = shallow(<Main pageNumber={3} {...props}/>);
    });

    it('renders the Page3 component when pageNumber=3', () => {
        expect(wrapper.find('Page3').length).toEqual(1);
    });

    it('renders main', () => {
        expect(wrapper.find('main.main').length).toEqual(1);
    });
});
