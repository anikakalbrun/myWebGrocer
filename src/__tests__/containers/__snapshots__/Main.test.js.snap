// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders the Page2 component when pageNumber=2 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "minHeight": 1109,
      "overflow": "auto",
    }
  }
>
  <ReduxForm />
</div>
`;

exports[`Main setComponentHeight works fine and is being called by windows resize 1`] = `
Object {
  "height": "auto",
  "minHeight": 1109,
  "overflow": "auto",
}
`;
